fsspec/implementations/prefix.py

```diff
@@ -0,0 +1,163 @@
+from pathlib import Path
+from typing import Any, Iterable, Sequence, Union
+
+import fsspec
+from fsspec import AbstractFileSystem
+from fsspec.core import split_protocol
+from fsspec.spec import AbstractBufferedFile
+from fsspec.utils import stringify_path
+
+
+class PrefixBufferedFile(AbstractBufferedFile):
+    def _fetch_range(self, start, end):
+        pass
+
+
+class PrefixFileSystem(AbstractFileSystem):
+    def __init__(
+        self,
+        prefix: str,
+        filesystem: fsspec.AbstractFileSystem,
+        *args,
+        **storage_options,
+    ) -> None:
+        super().__init__(*args, **storage_options)
+        self.prefix = stringify_path(prefix)
+
+        if not self.prefix:
+            self.prefix = self.sep
+
+        self.filesystem = filesystem
```

Comment:
How about
```suggestion
        self.fs = fs
```
same in the args, just makes it much easier to read in the rest of the code. Also `filesystem` is often used in this codebase for `fsspec.filesystem` helper, so will make grepping easier too.